CMakeLists.txt

```diff
@@ -68,6 +68,18 @@ if(ACL_CODE_COVERAGE)
   endif()
 endif()
 
+# Flag for tsan scans
+option(ACL_TSAN "Build with tsan" OFF)
+message(STATUS "Build with tsan: ${ACL_TSAN}")
+if(ACL_TSAN)
+  if(CMAKE_COMPILER_IS_GNUCXX)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ltsan -fsanitize=thread -g -O1 -fno-omit-frame-pointer -fPIC")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ltsan -fsanitize=thread -g -O1 -fno-omit-frame-pointer -fPIC")
```

Comment:
We use [`-Og` in `Debug` mode](https://github.com/intel/fpga-runtime-for-opencl/blob/8cd410fe9111a1fc9c7c122f54da1fc21c4fa587/CMakeLists.txt#L131-L133) now, would that be sufficient? I believe `-fPIC` is the default for all modes. `-g` is already provided by `Debug` mode; does it make much sense to support sanitizers in `Release` mode?

This is the corresponding snippet for ASAN:
https://github.com/intel/fpga-runtime-for-opencl/blob/8cd410fe9111a1fc9c7c122f54da1fc21c4fa587/CMakeLists.txt#L190-L201

Reply:
Thanks! I have put cmake changes in another PR: https://github.com/intel/fpga-runtime-for-opencl/pull/76 with the suggested change.